### PR TITLE
chore: update ts-jest

### PR DESCRIPTION
**Pull request by pczeglik-iohk**
In this PR I'm improving unit test run by:

- updating `ts-jest` from v28.0.7 to v29.1.2
- updating browserslist's db
- fixing local test command for cardano package

### CODE REVIEW

**Review by DominikGuzei — APPROVED** [review]

File: apps/browser-extension-wallet/test/jest.config.js [hunk 1]
```diff
@@ -48,9 +48,12 @@ module.exports = createJestConfig({
     '!src/utils/fake-api-request.ts'
   ],
   setupFilesAfterEnv: ['./test/jest.setup.js', 'jest-canvas-mock'],
-  globals: {
-    'ts-jest': {
-      tsconfig: './src/tsconfig.json'
-    }
+  transform: {
```
Comment (pczeglik-iohk) [inline apps/browser-extension-wallet/test/jest.config.js:51]: After ts-jest update configuration needs to be changed 
```
ts-jest[ts-jest-transformer] (WARN) Define `ts-jest` config under `globals` is deprecated. Please do
transform: {
    <transform_regex>: ['ts-jest', { /* ts-jest config goes here in Jest */ }],
},```

File: yarn.lock [hunk 2]
```diff
@@ -25072,17 +25072,10 @@ __metadata:
...
-  resolution: "caniuse-lite@npm:1.0.30001538"
-  checksum: 94c5d55757a339c7cc175f08a024671e2b4e7c04f130b1015793303d637061347efb6ad84447c3b8137333e742d150b8ad9672716bbf2482646c2e63a56f6c55
-  languageName: node
-  linkType: hard
-
-"caniuse-lite@npm:^1.0.30001565":
-  version: 1.0.30001572
-  resolution: "caniuse-lite@npm:1.0.30001572"
-  checksum: 7d017a99a38e29ccee4ed3fc0ef1eb90cf082fcd3a7909c5c536c4ba1d55c5b26ecc1e4ad82c1caa6bfadce526764b354608710c9b61a75bdc7ce8ca15c5fcf2
+"caniuse-lite@npm:^1.0.0, caniuse-lite@npm:^1.0.30001109, caniuse-lite@npm:^1.0.30001248, caniuse-lite@npm:^1.0.30001251, caniuse-lite@npm:^1.0.30001280, caniuse-lite@npm:^1.0.30001400, caniuse-lite@npm:^1.0.30001517, caniuse-lite@npm:^1.0.30001565":
```
Comment (pczeglik-iohk) [inline yarn.lock:25075]: Update browserslist db to get rid off this message 👇  from test run output

```
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
```

File: .github/workflows/build-dev-preview.yml [hunk 3]
```diff
@@ -26,7 +26,7 @@ jobs:
         env:
           AVAILABLE_CHAINS: 'Preprod,Preview,Mainnet'
           DEFAULT_CHAIN: 'Preprod'
-        run: yarn test:coverage --maxWorkers=2
+        run: yarn test:coverage --maxWorkers=2 --silent
```
Comment (pczeglik-iohk) [inline .github/workflows/build-dev-preview.yml:29]: Silence test run output logs to list only errors, ignore warnings which obstruct the most important information in logs.

Comment (pczeglik-iohk) [reply .github/workflows/build-dev-preview.yml:29]: If you are curious what this looks like now have a look at this logs: https://github.com/input-output-hk/lace/actions/runs/8327912326/job/22786711711

For improved version check `Build and Test` in this PR's pipeline

File: packages/cardano/package.json [hunk 4]
```diff
@@ -33,8 +33,8 @@
     "prepare": "ts-patch install -s",
     "prestart": "yarn build",
     "start": "node dist/index.js",
-    "test": "NODE_ENV=test jest -c ./test/jest.config.js",
-    "test:coverage": "yarn test --coverage",
+    "test": "NODE_ENV=test run -T jest -c ./test/jest.config.js",
```
Comment (pczeglik-iohk) [inline packages/cardano/package.json:36]: test run from local machine does not work
```
λ yarn test
command not found: jest
```

**Comment by pczeglik-iohk** [comment]
In test output logs there is still issue with code coverage which I would like to improve in a separate PR. It produces a lot of noise and for sure this issue should be addressed. Below you can find the error log:
```
Failed to collect coverage from /home/runner/work/lace/lace/packages/cardano/src/wallet/types.ts
ERROR: Invalid file coverage object, missing keys, found:data
STACK: Error: Invalid file coverage object, missing keys, found:data
    at assertValidObject (/home/runner/work/lace/lace/node_modules/istanbul-lib-coverage/lib/file-coverage.js:36:15)
    at new FileCoverage (/home/runner/work/lace/lace/node_modules/istanbul-lib-coverage/lib/file-coverage.js:78:9)
    at CoverageMap.addFileCoverage (/home/runner/work/lace/lace/node_modules/istanbul-lib-coverage/lib/coverage-map.js:109:21)
    at /home/runner/work/lace/lace/node_modules/@jest/reporters/build/CoverageReporter.js:279:33
    at async Promise.all (index 1)
    at async CoverageReporter._addUntestedFiles (/home/runner/work/lace/lace/node_modules/@jest/reporters/build/CoverageReporter.js:296:7)
    at async CoverageReporter.onRunComplete (/home/runner/work/lace/lace/node_modules/@jest/reporters/build/CoverageReporter.js:171:5)
    at async ReporterDispatcher.onRunComplete (/home/runner/work/lace/lace/node_modules/@jest/core/build/ReporterDispatcher.js:71:9)
    at async TestScheduler.scheduleTests (/home/runner/work/lace/lace/node_modules/@jest/core/build/TestScheduler.js:306:5)
    at async runJest (/home/runner/work/lace/lace/node_modules/@jest/core/build/runJest.js:367:19)
```

File: apps/browser-extension-wallet/test/jest.config.js (hunk 1 above)
Comment (mkazlauskas) [reply apps/browser-extension-wallet/test/jest.config.js:51]: Would be nice to import/extend common configuration to keep it DRY and consistent across packages, but not in scope of this chore, so not a change request :)

File: packages/cardano/package.json (hunk 4 above)
Comment (mkazlauskas) [reply packages/cardano/package.json:36]: Nice, I think I had the same issue locally before

**Review by mkazlauskas — APPROVED** [review]
Nice work! :rocket:

File: apps/browser-extension-wallet/test/jest.config.js (hunk 1 above)
Comment (pczeglik-iohk) [reply apps/browser-extension-wallet/test/jest.config.js:51]: will address this feedback in upcoming PR. Thank you

**Pull request merged**